### DISCUSSION TITLE
How to repack complete repository with defined pack size

### DISCUSSION THREAD

**Discussion by elgatobavaria** [rustic-rs/rustic#1169] [Q&A] [answered]
Hi guys,
i got a repo of about 2TB with current pack size of 128MB. I would to convert the repository to different pack sizes to see if it reduces the repo size. Is this possible with rustic ? If yes, how to do that ?

I already tried a little bit of that idea via restic and was not successfull: [Link to restic forum](https://forum.restic.net/t/does-repo-size-not-change-after-repacking/7869)

**Comment by aawsome** [comment, marked as answer]
Hi and welcome @elgatobavaria!

Thanks for your question! Can you tell me why you thing that the repository size would be (significantly) reduced by changing the pack size? IMO it should only differ marginally.

Besides this, changing the pack size of an existing repo can be done by:
- changing the repository target packsize using `rustic config` using the `--set-datapack-size*` options and `--set-*-packsize-tolerate-percent` options (read `rustic config -h` for information about the options)
- the next `prune` run will automatically repack non-fitting packs. You may want to use `--instant-delete` if you can ensure that you repo is not accessed in parallel.

**Comment by aawsome** [comment]
Note that rustic doesn't use a fixed (target) pack size, but by default increases the packsize automatically with growing repository size. The factors for this growing can be customized by the above options.

**Reply by elgatobavaria** [reply]
@aawsome I saw the behaveour when testing a little bit with restic --> see link in first post. So i thought i could reduce my current repo size by adapting the pack-size. 
Thank you for your quick and usefull answer.

**Reply by aawsome** [reply]
@elgatobavaria I very much doubt that you get such differences (69 GiB vs 37 GiB) by using different pack sizes - with either restic or rustic.
I would guess that either the backuped is different (i.e. different snapshots), or you used different compression parameters (for highly compressible data), or you have some unused but non-pruned blobs in the larger repository. 

But you might want to discuss this in the restic forum if you think you can get such results with restic. On contrast, if you succeed to get such results (by only changing the target pack size) with rustic, please post it! (but as already said I would be very much irritated if you can).

**Reply by elgatobavaria** [reply]
I can check it again on monday. But the test was executed automatically by a script so i would think that the data inside the repot must be equal. And compression was hard coded into the script.

**Comment by elgatobavaria** [comment]
@aawsome I checked the two repos with 8Mib and 128Mib pack size again. It seems that the script had not set compression to max. So i execute it again.
![restic_8](https://github.com/rustic-rs/rustic/assets/104026114/f6325e9a-77fb-4c93-ac1d-c86806b3a2e8)
![restic_128](https://github.com/rustic-rs/rustic/assets/104026114/8de4a352-5a26-4896-86bd-1d5a3d1e995b)

**Reply by aawsome** [reply]
@elgatobavaria I think your restic results are more interesting to the restic community - here we won't be able to say much about it.

If you are trying to reproduce some behavior in rustic, I'd suggest you use `rustic snapshots --long` and `rustic repoinfo` to get your information about usage and packsizes...

**Reply by elgatobavaria** [reply]
Executed the script again and verified that the compression ratio is set correct. Now the repo with pack size of 32 is the smallest with approx 32 GB. I will give rustic a try. 

Thank you !
